@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -347,13 +347,13 @@ export class TileGeometryLoader {
         tile.clear();
         // Set up techniques which should be processed.
         geometryCreator.initDecodedTile(decodedTile, enabledKinds, disabledKinds);
-        geometryCreator.createAllGeometries(tile, decodedTile);
-
-        if (stats.enabled) {
-            this.addStats(stats, now);
-        }
-        this.finish();
-        tile.dataSource.requestUpdate();
+        geometryCreator.createAllGeometries(tile, decodedTile).then(() => {
```

Comment:
I think it is nicer to mark the function `createGeometry` as async (from a documentation point of view). The added benefit is that you can await here.

Reply:
done